### PR TITLE
ADAPT-2142: Dismiss Button as its own component and Alert tweaks

### PR DESCRIPTION
**Pull request by yvonnetangsu**
# READY FOR REVIEW

# Summary
- Add Dismiss Button as its own component since we often need one in modals and other components.
- Use it in the Alert component

# Needed By (Date)
- When does this need to be merged by?

# Urgency
- How critical is this PR?

# Steps to Test

1. Look at code
1. Look at preview build and see the Dismiss Button examples
2. Look at the Alert component and see that alignment of labels, content, dismiss button should be improved. Also spacing has been tweaked - please check the different examples at different breakpoints.

# Affected Projects or Products
- Does this PR impact any particular projects, products, or modules?

# Associated Issues and/or People
- JIRA ticket
- Other PRs
- Any other contextual information that might be helpful (e.g., description of a bug that this PR fixes, new functionality that it adds, etc.)
- Anyone who should be notified? (`@mention` them here)

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

File: src/DismissButton/DismissButton.js [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+import Icon from 'react-hero-icon';
+import clsxd from 'clsx-dedupe';
+
+/**
+ * Dismiss Button Component
+ *
+ * For use with other components such as DismissButtons and modals.
+ *
+ */
+export const DismissButton = ({ classes = {}, text, srText, color, icon, iconType, customIcon, onClick, ...props }) => {
```
Comment (sherakama) [inline src/DismissButton/DismissButton.js:15]:
```suggestion
export const DismissButton = ({ className, text, srText, color, icon, iconType, iconClasses, customIcon, onClick, ...props }) => {
```

File: src/DismissButton/DismissButton.js [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+  const defaultIcon = <Icon icon={heroicon} type={heroiconType} aria-hidden='true' className={classes.icon} {...iconProps} />;
+
+  // Use custom icon for the component if it's provided; if not, use the default heroicon
+  const dismissIcon = customIcon ?? defaultIcon;
+
+  return (
+    <Button
+      variant='none'
+      size='minimal'
+      className={clsxd('su-flex su-items-center su-w-fit su-sans su-font-semibold su-leading-display', levers.color, classes.wrapper)}
```
Comment (sherakama) [inline src/DismissButton/DismissButton.js:61]:
```suggestion
      className={clsxd('su-flex su-items-center su-w-fit su-sans su-font-semibold su-leading-display', levers.color, className)}
```

File: src/DismissButton/DismissButton.js [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+  }
+
+  let heroiconType = 'solid';
+
+  if (iconType && dismissIconTypes.includes(iconType)) {
+    heroiconType = iconType;
+  }
+
+  // Set default heroicon if custom icon element isn't provided
+  const defaultIcon = <Icon icon={heroicon} type={heroiconType} aria-hidden='true' className={classes.icon} {...iconProps} />;
```
Comment (sherakama) [inline src/DismissButton/DismissButton.js:52]:
```suggestion
  const defaultIcon = <Icon icon={heroicon} type={heroiconType} aria-hidden='true' {...iconProps} />;
```

File: src/DismissButton/DismissButton.js [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+/**
+ * Dismiss Button Component
+ *
+ * For use with other components such as DismissButtons and modals.
+ *
+ */
+export const DismissButton = ({ classes = {}, text, srText, color, icon, iconType, customIcon, onClick, ...props }) => {
+  // Defaults & Variables
+  const levers = {};
+  const iconProps = { height: 20, width: 20 };
```
Comment (sherakama) [inline src/DismissButton/DismissButton.js:18]:
```suggestion
  iconProps = Object.assign({ height: 20, width: 20 }, ...iconProps);
```

File: src/DismissButton/DismissButton.js [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+        PropTypes.array
+      ]),
+      icon: PropTypes.oneOfType([
+        PropTypes.string,
+        PropTypes.object,
+        PropTypes.array
+      ]),
+    }
+  )
+};
```
Comment (sherakama) [inline src/DismissButton/DismissButton.js:130]:
```suggestion
```

File: src/DismissButton/DismissButton.js [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+  /**
+   * If you want to provide your own custom icon element
+   */
+  customIcon: PropTypes.element,
+
+  /**
+   * onClick function
+   */
+  onClick: PropTypes.func,
+
```
Comment (sherakama) [inline src/DismissButton/DismissButton.js:112]:
```suggestion

  iconProps: PropTypes.object,
```

**Review by sherakama — APPROVED** [review]
This looks great. I free edited on GH to show a point where you could have collisions on your `{...props}` and what gets passed into the Component. This is really a one level component with a child element so I don't think it needs the class object as a prop and we should be able to use the className convention. Also, if you put `...props` on one of the elements you should make sure you destructure out any other property you have set on that element first. eg:

`<p id="12" className={clsxd("thing", className)} {...props}>children</p>`

You should destructure `id` and `className` out of the props so that developers don't actually duplicate something you are assigning by accident.

**Review by sherakama — CHANGES_REQUESTED** [review]
Meant to request changes.

**Comment by sherakama** [comment]
Also, the timing on this is amazing as I will likely need it for the error message to the user on failed MegaProfile API calls and other messages.

**Comment by yvonnetangsu** [comment]
> This looks great. I free edited on GH to show a point where you could have collisions on your `{...props}` and what gets passed into the Component. This is really a one level component with a child element so I don't think it needs the class object as a prop and we should be able to use the className convention. Also, if you put `...props` on one of the elements you should make sure you destructure out any other property you have set on that element first. eg:
> 
> `<p id="12" className={clsxd("thing", className)} {...props}>children</p>`
> 
> You should destructure `id` and `className` out of the props so that developers don't actually duplicate something you are assigning by accident.

Oh nice - thank you for the review. Good point. I'll merge in your changes and make sure it's taken care of everywhere.

**Review by sherakama — APPROVED** [review]
GTG

**Comment by sherakama** [comment]
Thanks for doing this work!

**Pull request merged**